### DISCUSSION TITLE
Add Dataframe

### DISCUSSION THREAD

**Discussion by xingularity** [solvcon/modmesh#339] [General]
Add a DataFrame facility. A DataFrame is a two-dimensional, size-mutable tabular data structure with labeled axes (rows and columns). This DataFrame should have following capabilities.

1. Store a size-mutable tabular data.
2. With single row and column index (unsigned integer type).
3. Support limited data types first, primarily numerical.
4. Sub DataFrame selection and assignment by index.
5. Sorting data according to specific column and specify order.
6. Search data in DataFrame and return matched index.

**Comment by yungyuc** [comment]
Thanks @xingularity .  The list includes some features but I did not see any analytical calculations like average, variation, etc.  Could you think of some as a starting point for development?

**Comment by xingularity** [comment]
Hi @yungyuc I thin we can start with the following 6 functions.
* count
* mean
* std
* min
* max
* quantile (provide capabilities to get any quantile)

**Comment by yungyuc** [comment]
@xingularity I convert the issue into a discussion conversation because the scope is too large to be planned.  More concrete tasks can be generated from the conversation and we will track them.

A project is tracking the development: https://github.com/orgs/solvcon/projects/6/views/1?pane=info

**Reply by yungyuc** [reply]
@tigercosmos Could you please also take a look at this conversation and comment?

**Reply by tigercosmos** [reply]
I can help with this development, but I think I will need to study some background knowledge first. 🤔

**Reply by tigercosmos** [reply]
@yungyuc  btw, the url is broken.

**Reply by yungyuc** [reply]
> I can help with this development, but I think I will need to study some background knowledge first. 🤔

Let's use more discussion for what to do

**Reply by yungyuc** [reply]
> @yungyuc btw, the url is broken.

I can open it after clicking

**Reply by tigercosmos** [reply]
@xingularity do you have a picture of how to implement it? I think I can follow your design.

**Comment by yungyuc** [comment]
An old request discussing this can be found in https://github.com/solvcon/modmesh/issues/28
